autoload/dirvish.vim

```diff
@@ -86,7 +90,12 @@ endif
 func! s:list_dir(dir) abort
   let s:rel = get(g:, 'dirvish_relative_paths', 0)
   " Escape for globpath().
-  let dir_esc = escape(substitute(a:dir,'\[','[[]','g'), ',;*?{}^$\')
+  let char_esc = ',;*?{}^\'
+  if !has('win32')
+    " '$' is valid in win32 UNC path; and does not require escape in win32 globpath().
```

Comment:
Example: `globpath('\\wsl$\fedora\', '*', 0, 1)` is ok; but `globpath('\\wsl\$fedora\', '*', 0, 1)` will not work.